storage/minifile/mimifile.go

```diff
@@ -0,0 +1,307 @@
+package minifile
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"sync/atomic"
+
+	"github.com/prometheus/tsdb/fileutil"
+	"github.com/syndtr/goleveldb/leveldb/util"
+)
+
+type MiniFile struct {
+	path         string
+	instanceLock fileutil.Releaser // File-system lock to prevent double opens
+	namedLock    *NamedLock
```

Comment:
It seems that only one lock is needed? When calling the get/put/delete/update methods, the first step is to get the lock.

Reply:
namedlock is removed.